frontend/src/pages/MetricCreateEdit.jsx

```diff
@@ -0,0 +1,181 @@
+        object: data.object
+    }
+    console.log(responseData);
+  }
+  
+  const editMetric = async () => {
+    
+    const metricName = metricNameRef.current;
+    const metricType = metricTypeRef.current;
+    const metricDesc = metricDescRef.current; 
+
+    let formData = {
+        name: metricName.value,
+        type: metricType.value,
+        area: tags,
+        description: metricDesc.value
+    }
+
+    var data = await fetch(`http://127.0.0.1:8000/api/v1/metrics/${metric_id}`, {
+        method: 'PUT',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(formData),
+    })
+    .then((response) => response.json())
+    var responseData = {
+        message: data.message,
+        object: data
+    }
+    console.log(responseData)
+  }
+
+  const getMetricById = async (metric_id) => {
+    try{
+        var data = await fetch(`http://127.0.0.1:8000/api/v1/metrics/${metric_id}`)
+        .then((response) => response.json());
+        
+        if( data.message == "Not Found") {
+            // replace path history to a 404 not found page
+            navigate('/404');
+        }
+
+        var metric_data = data;
+        console.log(metric_data);
+        
+        const metricName = metricNameRef.current;
+        const metricType = metricTypeRef.current;
+        const metricDesc = metricDescRef.current; 
+
+        metricName.value = metric_data.name;
+        metricType.value = metric_data.type;
+        metricDesc.value = metric_data.description;
+
+        setTags([...metric_data.area]);
+
+    } catch (error) {
+        console.log(error);
+    }
+
+  };
+
+  const handleFormClick = async (e) => {
+    e.preventDefault();
+    if (props.mode == "create"){
+        await postMetric();
+    }
+    else if (props.mode == "edit"){
+        await editMetric(metric_id);
+    }
+    navigate('/dashboard/metrics', {state: { forceRender: true }});
+  }
+
+  let disable = 'false'
+
+  useEffect(()=> {
+    if(props.mode == 'edit'){
+        getMetricById(metric_id)
+        if(props.mode == 'view'){
+            disable = 'true'
+        }
+    }
+  }, [metric_id])
+
+  return (
+    <Layout>
+        <div className="content mt-4 mb-5" style={{width: '100%'}}>
+            <div className="container">
+                <div className="row">
+                    <div className="col-sm-6 m-auto mb-5">
+                        <h1 className="text-capitalize">{props.mode} Metric</h1>
+                        <form data-mode="create" className="row g-3" id="metric-form"
+                        onKeyDown={(e)=> {return e.key !== 'Enter'}}
+                        onSubmit={(e) => e.preventDefault()}
+                        >
+                            <div className="col-12">
+                                <label htmlFor="metric-name" className="form-label">Metric Name</label>
+                                <input ref={metricNameRef} type="text" className="form-control"
```

Comment:
why the excessive usage of refs while you can simply define states and dynamically bind them to the value of the elements?
const [metricName, setMetricName] = useState("");

const handleMetricNameChange = useCallback((metricName: string) => {
// perform any validation on the value.

setMetricName(metricName);
}, []);

<input  type="text" className="form-control" placeholder="" id="metric-name" value={metricName} onChange={handleMetricNameChange} />

so when you want to update the value of the input element simply call the setMetricName with the value you want.

Reply:
the input element becomes `<input type="text" className="form-control"  placeholder="" id="metric-name" value={metricName} onChange={(e) => handleMetricNameChange(e.target.value)} />`

Reply:
Done